crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -554,7 +567,7 @@ impl Context {
                         }
                     };
                     if index >= array_size {
-                        // Ignore the error if side effects are disabled.
+                        // Report the error if side effects are disabled.
```

Comment:
```suggestion
                        // Report the error if side effects are enabled.
```
Is this not the case here?

Reply:
Done